### PR TITLE
3.6 into main

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Forward merge:
- https://github.com/juju/juju/pull/17348
- https://github.com/juju/juju/pull/17413
- https://github.com/juju/juju/pull/17423
- https://github.com/juju/juju/pull/17436
- https://github.com/juju/juju/pull/17336
- https://github.com/juju/juju/pull/17422
- https://github.com/juju/juju/pull/17446
- https://github.com/juju/juju/pull/17448
- https://github.com/juju/juju/pull/17444
- https://github.com/juju/juju/pull/17453
- https://github.com/juju/juju/pull/17440
- https://github.com/juju/juju/pull/17458
- https://github.com/juju/juju/pull/17454
- https://github.com/juju/juju/pull/17461
- https://github.com/juju/juju/pull/17462
- https://github.com/juju/juju/pull/17463
- https://github.com/juju/juju/pull/17465
- https://github.com/juju/juju/pull/17469
- https://github.com/juju/juju/pull/17144
- https://github.com/juju/juju/pull/17179
- https://github.com/juju/juju/pull/17464
- https://github.com/juju/juju/pull/17476
- https://github.com/juju/juju/pull/17477

Conflicts:
- cmd/juju/application/bundle/bundle.go
- cmd/juju/application/bundle/bundle_test.go
- cmd/juju/application/deploy.go
- cmd/juju/application/deploy_test.go
- cmd/juju/commands/bootstrap.go
- cmd/juju/machine/add.go
- cmd/jujud-controller/agent/machine.go
- cmd/jujud/agent/machine.go
- go.mod
- go.sum
- internal/bundle/changes/changes.go
- internal/container/utils.go
- internal/container/utils_test.go
- internal/provider/azure/environ.go
- internal/provider/azure/internal/errorutils/errors_test.go
- internal/provider/azure/internal/imageutils/images.go
- internal/provider/azure/internal/imageutils/images_test.go
- internal/worker/containerbroker/broker_test.go
- internal/worker/uniter/runner/context/context.go
- internal/worker/uniter/runner/context/contextfactory.go
- internal/worker/uniter/runner/context/export_test.go
- internal/worker/uniter/runner/jujuc/relation-ids.go
- tests/includes/juju.sh
- tests/main.sh
- tests/suites/deploy/os.sh

Most conflicts were trivial to solve. Azure conflicts took a bit of thought. They emerged from [my PR](https://github.com/juju/juju/pull/17440)

The conflicts to `internal/worker/uniter/runner`, emerging from [this PR](https://github.com/juju/juju/pull/17464), took some thought, since the structure of tests had changed to mocks. The conflicts with the production code here, however, was trivial to merge

### CODE REVIEW

File: tests/suites/manual/deploy_manual.sh [hunk 1]
```diff
@@ -10,10 +10,10 @@ test_deploy_manual() {
 		cd .. || exit
 
 		case "${BOOTSTRAP_PROVIDER:-}" in
-		"lxd" | "lxd-remote" | "localhost")
+		"lxd")
 			run "run_deploy_manual_lxd"
 			;;
-		"aws" | "ec2")
```
Comment (manadart) [inline tests/suites/manual/deploy_manual.sh:16]: This seems to have been changed to `ec2` everywhere else...

**Review by manadart — COMMENTED** [review]
Have we got this right?

File: tests/suites/manual/deploy_manual.sh (hunk 1 above)
Comment (jack-w-shaw) [reply tests/suites/manual/deploy_manual.sh:16]: https://github.com/juju/juju/pull/17448/files#diff-022e60c5ac1d2ca17b86164b81b834a6b30c57a7e47950e26a9bc3a463fc3ec5L16

You'll have to ask Harry

Comment (jack-w-shaw) [reply tests/suites/manual/deploy_manual.sh:16]: But agreed, this looks wrong

**Comment by jack-w-shaw** [comment]
> Looks like we need a `go mod tidy` too.

Done

**Review by manadart — APPROVED** [review]
Thanks for this.

**Pull request merged**